### PR TITLE
[do not merge] Exercise: add outer margins

### PR DESCRIPTION
**Pull request by kaelig**
Because o-grid is pretty complex, I built an exercise for you @AlbertoElias, so that you can understand the code and tests.

At the moment, outer margins (space on the left and right of a o-grid-container) are the same as gutters: 10px < M, and 20px on layouts >= M.

It works exactly like for gutters.

This branch contains a few tests. To make them pass here is what is needed:
- add an `$o-grid-outer-margins` map that contains outer margins for each breakpoint
- modify the mixin `oGridAddLayout` so that developers can add a layout with a specific `$outer-margin-width`:
    
    ```scss
    @mixin oGridAddLayout($layout-name, $layout-width, $gutter-width: null, $outer-margin-width: null) {
    ```

- Create an `oGrid.getCurrentOuterMargin` JS helper function

ps: not sure that should be merged when you're done, since it's not something anybody needed. But after doing this exercise you should know all the secrets behind the grid!

### CODE REVIEW

File: src/scss/_main.scss [hunk 1]
```diff
@@ -12,22 +12,26 @@
...
+/// @param {Number} $outer-margin-width [null] - Outer margin width in px
+@mixin oGridAddLayout($layout-name, $layout-width, $gutter-width: null, $outer-margin-width: null) {
 	$temp-layouts: ();
 	$temp-gutters: (default: #{oGridGutter()});
+	$temp-outer-margins: (default: #{oGridGutter()});
 
 	// Add the new layout in the correct position:
-	// (we want $o-grid-layouts and $o-grid-gutters to ordered from the smallest to the largest layout)
+	// (we want $o-grid-layouts and $o-grid-gutters and $o-grid-outer-margins to be ordered from the smallest to the largest layout)
 	@for $index from 1 through length($o-grid-layouts) {
```
Comment (AlbertoElias) [inline src/scss/_main.scss]: @kaelig my only question is about default, why is not overrideable?

**Comment by AlbertoElias** [comment]
This is now done. I'm going to close this, but I'll keep the branch open in case anyone else wants to give it a go.Though do checkout the commit Kaelig made!

File: src/scss/_main.scss (hunk 1 above)
Comment (kaelig) [inline src/scss/_main.scss]: Good question, should it be?

**Comment by kaelig** [comment]
Well done! Did it help you understand the code base better?

File: src/scss/_main.scss (hunk 1 above)
Comment (AlbertoElias) [inline src/scss/_main.scss]: No use case currently, but if it was the opposite, we probably would be saying the same about it making it not overrideable, but it takes a bit more effort to not make it overrideable. I was just wondering if there was a reason

**Comment by AlbertoElias** [comment]
Yeah, it was pretty good, thanks for that!

**Pull request closed without merging**